pkg/router/controller/router_controller.go

```diff
@@ -321,3 +194,61 @@ func (c *RouterController) processIngressEvents(events []ingressRouteEvents) {
 		}
 	}
 }
+
+func (c *RouterController) handleFirstSync() {
+	if err := c.WaitForInformerCacheSync(); err != nil {
+		glog.Fatalf("Failed to sync router informer cache: %v", err)
+	}
+	c.processExistingItems()
+	c.firstSyncDone = true
```

Comment:
The only part of this method that really belongs here is setting this Boolean (which needs to be under a lock).